src/main/java/teammates/common/datatransfer/InstructorAttributes.java

```diff
@@ -324,18 +314,9 @@ public boolean hasTutorPrivileges() {
         return privileges.hasTutorPrivileges();
     }
 
-    /**
-     * pre-condition: instructorPrivilegesAsText and privileges should be non-null
-     * @param instructor
-     * @return
-     */
     public boolean isEqualToAnotherInstructor(InstructorAttributes instructor) {
-        if (gson.toJson(this).equals(gson.toJson(instructor))) {
-            return true;
-        } else {
-            return !instructorPrivilegesAsText.equals(instructor.instructorPrivilegesAsText) 
-                   && privileges.equals(instructor.privileges);
-        }
+        JsonParser parser = new JsonParser();
+        return parser.parse(getJsonString()).equals(parser.parse(instructor.getJsonString()));
```

Comment:
Not exactly, `parser.parse(getJsonString())` parses the string and stores the variable names and values of `InstructorAttributes` in some data structures within `JsonElement`. `fromJson(getJsonString(), JsonElement.class)` attempts to create a `JsonElement` object  from the string returned by `getJsonString()`. This cannot be done correctly because string does not have the variables of `JsonElement`.